FPCO-14583: StepperView

rushangfynd opened the pull request:
- Stepper view configuration with increment and decrement operation 
- High level configuration for style and theme.

Commit:
iOS Stepper view component with + and - operation

Remove test image for checking asset type

Rebase branch with fork master

rushangfynd committed:
Merge branch 'master' into feature/FPCO-14583-stepper-control

rushangfynd committed:
Update README.md

rushangfynd committed:
Update README.md

hitendra-gofynd commented on Sources/Nitrozen-SwiftUI/Elements/Stepper/NitrozenStepperView.swift:
```diff
@@ -0,0 +1,163 @@
+    
+    @Binding var inputValue:String
+    var appearance: NitrozenAppearance.StepperView
+    var maxNumber:Int
```
do we have option to limit the min number??  Always start with Apple provided native APIs when you start working on anything that is already implement.

Check below code how we can use iOS Provided stepper.
`struct StepperView: View {
    @State private var value = 0
    let step = 5
    let range = 1...50

    var body: some View {
        Stepper(value: $value,
                in: range,
                step: step) {
            Text("Current: \(value) in \(range.description) " +
                 "stepping by \(step)")
        }
            .padding(10)
    }
}`

Ref: https://developer.apple.com/documentation/swiftui/stepper

hitendra-gofynd replied:
Tip: always think about all the possible use cases and try to predict universal usage of any solution you are providing -- that will make you real Software Engineer. There is a big difference between Software Developer vs Software Engineer -- As a team, I want to see you @rushangfynd and @keyur-gofynd as Software Engineers and not only developers.

hitendra-gofynd commented on Sources/Nitrozen-SwiftUI/Elements/Stepper/NitrozenStepperView.swift:
```diff
@@ -0,0 +1,163 @@
+    func increment(){
+        if Int(inputValue).or(0) < maxNumber{
+            var value = Int(inputValue).or(0)
+            value += 1
```
Also consider a case where use needs to increment step value by 5,
e.g. start with 10 and end with 100 i.e. 10...100, and step will be incremented by 5 -- i.e. 10-15-20-25-....100

rushangfynd committed:
Step and range support implement for stepper

rushangfynd committed:
Merge branch 'master' into feature/FPCO-14583-stepper-control

hitendradeveloper commented on Example-Nitrozen-SwiftUI/Example-Nitrozen-SwiftUI/Example-Code/Stepper.swift:
```diff
@@ -0,0 +1,102 @@
+                Section("Custom button for steper 10 Limit\ncurrent count \(countText5)"){
+                    NitrozenStepperView(
+                        inputValue: $countText5,
+                        appearance: NitrozenAppearance.shared.stepperView.copy
```
Move appearance to last parameter

hitendradeveloper commented on Example-Nitrozen-SwiftUI/Example-Nitrozen-SwiftUI/Example-Code/UIElementsList.swift:
```diff
@@ -9,167 +9,187 @@ import SwiftUI
 import Nitrozen_SwiftUI
 
 struct UIElementsList: View {
```
Verify content of this file -- many lines removed and added for same code 

-- verify by matching spacing setttigs of Keyur -- later we will sync all of us settings

hitendradeveloper commented on Sources/Nitrozen-SwiftUI/Elements/Stepper/NitrozenStepperView.swift:
```diff
@@ -0,0 +1,196 @@
+    }
+    
+    func increment(){
+        if Int(inputValue).or(0) < maxInputValue{
```
Instead of converting to Int multiple times, declare variable
intInputValue

hitendradeveloper commented on Sources/Nitrozen-SwiftUI/Elements/Stepper/NitrozenStepperView.swift:
```diff
@@ -0,0 +1,196 @@
+            var value = Int(inputValue).or(0)
+            value += step
+            if value > self.maxInputValue {
+                inputValue = min(value, self.maxInputValue).description
```
This line will always works, if/else condition not required-- try testing removing if/else

rushangfynd replied:
Done

hitendradeveloper commented on Sources/Nitrozen-SwiftUI/Elements/Stepper/NitrozenStepperView.swift:
```diff
@@ -0,0 +1,196 @@
+        
+    }
+    
+    func decrement(){
```
Same comments like + button --- refactor the code

rushangfynd replied:
code modification: 

```
        if minInputValue < intInputValue{
            var value = intInputValue
            value -= step
            inputValue = max(value, self.minInputValue).description
        }
        else{
            inputValue = max(Int(inputValue) ?? 00, self.minInputValue).description
        }

```

Here we required else  

Case: If user enter the smaller value and try to decrease then it we required to setup lowerRange value forcefully.

lowerRange: 1000 
Entered Value: 100 

then it will be 1000 forcefully

hitendra-gofynd replied:
use min and max combination here in that case to keep value in range

hitendradeveloper reviewed (changes requested):
Please fix mentioned comments

rushangfynd committed:
Merge request comment fix

rushangfynd committed:
Comparison space fix in UIElement and NitrozenAppearance file

Format change to tab in xcode setting

Formate of file setup as per team setting

hitendra-gofynd commented on Sources/Nitrozen-SwiftUI/Elements/Stepper/NitrozenStepperView.swift:
```diff
@@ -0,0 +1,194 @@
+    
+    public var body: some View {
+        HStack(spacing: self.itemSpacing){
+            decrementAction()
```
rename to decrementActionButton()
decrementAction() indicates execution of some action

hitendra-gofynd commented on Sources/Nitrozen-SwiftUI/Elements/Stepper/NitrozenStepperView.swift:
```diff
@@ -0,0 +1,194 @@
+        HStack(spacing: self.itemSpacing){
+            decrementAction()
+            inputTextfield()
+            incrementAction()
```
same like above comment

rushangfynd committed:
Condition improve for increment and decrement

hitendra-gofynd commented on Sources/Nitrozen-SwiftUI/Elements/Stepper/NitrozenStepperView.swift:
```diff
@@ -0,0 +1,166 @@
+    
+    public var body: some View {
+        HStack(spacing: self.itemSpacing){
+			decrementActionButton()
```
indentation of this line
